libs/pageserver_api/src/models.rs

```diff
@@ -18,7 +18,29 @@ use crate::reltag::RelTag;
 use anyhow::bail;
 use bytes::{BufMut, Bytes, BytesMut};
 
-/// A state of a tenant in pageserver's memory.
+/// The state of a tenant in this pageserver.
+///
+/// ```mermaid
+/// stateDiagram-v2
+///
+///     [*] --> Loading: spawn_load()
+///     [*] --> Attaching: spawn_attach()
+///
+///     Loading --> Activating: activate()
+///     Attaching --> Activating: activate()
+///     Activating --> Active: infallible
+///
+///     Loading --> Broken: load() failure
+///     Attaching --> Broken: attach() failure
+///
+///     Active --> Stopping: set_stopping(), part of shutdown & detach
+///     Stopping --> Broken: late error in remove_tenant_from_memory
+///
+///     Broken --> [*]: ignore / detach / shutdown
+///     Stopping --> [*]: remove_from_memory complete
+///
+///     Active --> Broken: cfg(testing)-only tenant break point
+/// ```
```

Comment:
This can be viewed with https://mermaid.live at least. Unsure if this is the way to go.

Reply:
> Well, apparently support exists: https://github.com/mersinvald/aquamarine

Yeah, I was afraid of the review comments for pulling that in ;)

Reply:
Also regex version would work and not require an external thing to view.

Reply:
What is this "regex version" you are talking about?

Reply:
Before your changes: https://github.com/neondatabase/neon/pull/4321#discussion_r1205238263

Reply:
I don't want to to block this PR on mermaid.